examples/adversarial/test_hans.py

```diff
@@ -154,10 +158,14 @@ def train(args, train_dataset, model, tokenizer):
         for step, batch in enumerate(epoch_iterator):
             model.train()
             batch = tuple(t.to(args.device) for t in batch)
-            inputs = {"input_ids": batch[0], "attention_mask": batch[1], "labels": batch[3]}
+            inputs = {
+                "input_ids": batch["input_ids"],
+                "attention_mask": batch["attention_mask"],
+                "labels": batch["labels"],
+            }
             if args.model_type != "distilbert":
                 inputs["token_type_ids"] = (
-                    batch[2] if args.model_type in ["bert", "xlnet"] else None
+                    batch["token_type_ids"] if args.model_type in ["bert", "xlnet"] else None
                 )  # XLM, DistilBERT and RoBERTa don't use segment_ids
```

Comment:
This can now be way simpler. Models that don't use `token_type_ids` don't output it anymore (I believe XLM makes use of them, but I may be wrong).

Therefore by using `tokenizer.encode_plus`, you can obtain the model-specific output right away.

e.g. with BERT:

```py
bert_tokenizer.encode_plus("hey, how are you?")
# {'input_ids': [101, 23998, 117, 1293, 1132, 1128, 136, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1]}
```

whereas with DistilBERT:

```py
distilbert_tokenizer.encode_plus("hey, how are you?")
# {'input_ids': [101, 23998, 117, 1293, 1132, 1128, 136, 102], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1]}
```

This is so that you can chain the two methods:

```py
inputs = tokenizer.encode_plus(xxx)
model(**inputs)
```